### PR TITLE
Add percona 8.1 package, role and test

### PR DESCRIPTION
**Pull request by snglth**
@flyingcircusio/release-managers

percona-xtrabackup's generic.nix did not work to build percona-xtrabackup 8.1 since it was using percona-xtrabackup 2.4 source from binary cache for some reason

## Release process

Impact:

Changelog:

### PR release workflow (internal)

- [ ] PR has internal ticket
- [ ] internal issue ID (PL-…) part of branch name
- [ ] internal issue ID mentioned in PR description text
- [ ] ticket is on Platform agile board
- [ ] ticket state set to *Pull request ready*
- [ ] if ticket is more urgent than within the next few days, directly contact a member of the Platform team

## Design notes

- [ ] Provide a feature toggle if the change might need to be adjusted/reverted quickly depending on context. Consider whether the default should be `on` or `off`. Example: rate limiting.
- [ ] All customer-facing features and (NixOS) options need to be discoverable from documentation. Add or update relevant documentation such that hosted and guided customers can understand it as well.

## Security implications

- [ ] [Security requirements](https://wiki.flyingcircus.io/System_Development_Guideline#Security_requirement_principles_and_testing) defined? (WHERE)
- [ ] Security requirements tested? (EVIDENCE)

### CODE REVIEW

**Comment by dpausp** [comment]
Thanks for the PR! I'll have a closer look later this week.

File: pkgs/percona-xtrabackup/8_1.nix [hunk 1]
```diff
@@ -0,0 +1,55 @@
+{ lib, stdenv, fetchurl, bison, boost, cmake, makeWrapper, pkg-config
+, curl, cyrus_sasl, libaio, libedit, libev, libevent, libgcrypt, libgpg-error, lz4
+, ncurses, numactl, openssl, procps, protobuf, valgrind, xxd, zlib
+, perlPackages
+, fetchSubmodules ? false, extraPatches ? [], extraPostInstall ? "", ...
+}:
+
+stdenv.mkDerivation rec {
```
Comment (osnyx) [inline pkgs/percona-xtrabackup/8_1.nix]:
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

Modern nixpkgs prefer this overlay-style way of referring to internal attributes, as this composes much better with overriding these attributes from the outside later.

File: pkgs/percona-xtrabackup/8_1.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  pname = "percona-xtrabackup";
+  version = "8.1.0-1";
+
+  src = fetchurl {
+    url = "https://downloads.percona.com/downloads/Percona-XtraBackup-innovative-release/Percona-XtraBackup-${version}/source/tarball/percona-xtrabackup-${version}.tar.gz";
```
Comment (osnyx) [inline pkgs/percona-xtrabackup/8_1.nix]:
```suggestion
    url = "https://downloads.percona.com/downloads/Percona-XtraBackup-innovative-release/Percona-XtraBackup-${finalAttrs.version}/source/tarball/percona-xtrabackup-${finalAttrs.version}.tar.gz";
```

see above

File: pkgs/percona-xtrabackup/8_1.nix [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  '' + extraPostInstall;
+
+  meta = with lib; {
+    description = "Non-blocking backup tool for MySQL";
+    homepage = "http://www.percona.com/software/percona-xtrabackup";
+    license = licenses.lgpl2;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ izorkin ];
+  };
+}
```
Comment (osnyx) [inline pkgs/percona-xtrabackup/8_1.nix]:
```suggestion
})
```

see above

**Review by osnyx — APPROVED** [review]
Thank you for your PR. Overall, your changes are fine when it comes to the goal of adding a new percona version. I only had a slight nitpick regarding the usage of `mkDerivation`, see the comments.

Incidentally though I have discovered that we've already moved to pulling percona from upstream nixpkgs (which we also maintain there) and thus rebased your changes onto the Nix expression from there.

In general, we prefer to work upstream-first, so in the future you *might* do such PRs against nixpkgs and alert us to take care of this. But if you're not interested in this and just want to get something usable on our platform, of course we can take care of this, as we'll do with this PR, and continue to do PRs against fc-nixos for simplicity's sake. =)

For transparency reasons, I've kept the changes as additional commits so far but would like to squash some of them before merging tomorrow.

**Comment by osnyx** [comment]
## Release process

PL-132430

Impact: none

Changelog:
- percona: add package and full role for percona-8.1 as `flyingcircus.roles.percona81`

### PR release workflow (internal)

- [x] PR has internal ticket
- [ ] internal issue ID (PL-…) part of branch name
- [x] internal issue ID mentioned in PR description text
- [x] ticket is on Platform agile board
- [x] ticket state set to *Pull request ready*
- [ ] if ticket is more urgent than within the next few days, directly contact a member of the Platform team

## Design notes

- [x] Provide a feature toggle if the change might need to be adjusted/reverted quickly depending on context. Consider whether the default should be `on` or `off`. Example: rate limiting.
- [x] All customer-facing features and (NixOS) options need to be discoverable from documentation. Add or update relevant documentation such that hosted and guided customers can understand it as well.

## Security implications

- [x] [Security requirements](https://wiki.flyingcircus.io/System_Development_Guideline#Security_requirement_principles_and_testing) defined? (WHERE)
  - [x] must not introduce any known regressions
- [x] Security requirements tested? (EVIDENCE)
  - [x] expanded automated tests
  - [x] automated tests still pass
  - [x] deliberately accepting that percona-8.1 is already EOL as the customer still intends to use that specific version

**Pull request merged**